proxy/http2/Http2ClientSession.cc

```diff
@@ -272,7 +272,10 @@ Http2ClientSession::do_io_close(int alerrno)
     client_vc->do_io_close();
     client_vc = nullptr;
   }
-  this->connection_state.release_stream(nullptr);
+  {
```

Comment:
I know it's benign (and correct), but do we really need the extra closure / scope here? Just getting rid of it would have the same behavior, since it will go out of scope the same way. I just find the above uglier, and possible confusing :-).

Reply:
I added the extra scope because I wanted to minimize the risk of dead lock. To make it readable, I added an empty line right before the scope.